kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/denali.java

```diff
@@ -140,30 +140,18 @@ public void onFailure(Throwable caught) {
 
                                     @Override
                                     public void onSuccess(ArrayList<GwtGroupedNVPair> pairs) {
-                                        denali.this.m_connected = false;
+                                        denali.this.connected = false;
                                         int connectionNameIndex = 0;
                                         for (GwtGroupedNVPair result : pairs) {
                                             if ("Connection Name".equals(result.getName())
-                                                    && "CloudService".equals(result.getValue())) {
-                                                GwtGroupedNVPair connectionStatus = pairs.get(connectionNameIndex + 1); // done
- // based
- // on
- // the
- // idea
- // that
- // in
- // the
- // pairs
- // data
- // connection
- // name
- // is
- // before
- // connection
- // status
-                                                if ("Connection Status".equals(connectionStatus.getName())
+                                                    && result.getValue().endsWith("CloudService")) {
+                                                // done based on the idea that in the pairs data connection name is
```

Comment:
``// based on the assumption the Service Status pair follows the Connection Name pair``